.buildkite/custom-tests.json

```diff
@@ -0,0 +1,82 @@
+{
+  "tests": [
+    {
+      "test_name": "staging: build-gnu",
+      "command": "cd staging && RUSTFLAGS=\"-D warnings\" cargo build --release",
+      "soft_fail": "true",
+      "platform": [
+        "x86_64",
+        "aarch64"
+      ]
+    },
+    {
+      "test_name": "staging: build-musl",
+      "command": "cd staging && RUSTFLAGS=\"-D warnings\" cargo build --release --target {target_platform}-unknown-linux-musl",
+      "soft_fail": "true",
+      "platform": [
+        "x86_64",
+        "aarch64"
+      ]
+    },
+    {
+      "test_name": "staging: style",
+      "command": "cd staging && cargo fmt --all -- --check --config format_code_in_doc_comments=true",
+      "soft_fail": "true"
+    },
```

Comment:
Hmm, maybe we can skip the `soft_fail` flag here? We should always make sure tests pass ?

Reply:
agree!